x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/plugin/RestSqlTranslateAction.java

```diff
@@ -47,8 +51,22 @@ public RestSqlTranslateAction(Settings settings, RestController controller) {
     protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) throws IOException {
         SqlTranslateRequest sqlRequest;
         try (XContentParser parser = request.contentOrSourceParamParser()) {
-            sqlRequest = SqlTranslateRequest.fromXContent(parser, Mode.fromString(request.param("mode")));
+            sqlRequest = SqlTranslateRequest.fromXContent(parser);
```

Comment:
This block is repeated here: https://github.com/elastic/elasticsearch/pull/36149/files#diff-47894e3a968e76739b45d73c3405ab8eR61 and here: https://github.com/elastic/elasticsearch/pull/36149/files#diff-33543baa55df79bc3941f885b6fcfa61R47 so maybe extract it to a method.